packages/gatsby/src/schema/data-tree-utils.js

```diff
@@ -14,24 +23,36 @@ const flatten = require(`flat`)
  *
  * @param {*Nodes} args
  */
-export const extractFieldExamples = (nodes: any[]) => {
+const extractFieldExamples = (nodes: any[]) => {
   // $FlowFixMe
-  return _.mergeWith({}, ...nodes, (obj, next) => {
-    if (!_.isArray(obj || next)) return
-    let array = [].concat(obj, next).filter(v => v != null)
+  return _.mergeWith({}, ...nodes, (obj, next, key, o, s, stack) => {
+    if (obj === INVALID_VALUE) return obj
+
+    // TODO: if you want to support infering Union types this should be handled
+    // differently. Maybe merge all like types into examples for each type?
+    // e.g. union: [1, { foo: true }, ['brown']] -> Union Int|Object|List
+    if (!isSameType(obj, next)) return INVALID_VALUE
```

Comment:
actually as I'm thinking about it, its tough to throw...A plugin author would most likely "fix" this during the `extendNodes` phase which is after example gathering and not accessible for input inference. 

In the specific case here the union type is `component.props[].type.value` You could "fix" that in the node creation bit, but actually I just want to make that JSON scalar rather then split that out into 5 different fields, `arrayValue`, `enumValue`, `shapeValue`, `customValue`, etc. For one it differs fairly dramatically from the underlying source format and would require a bunch of documentation apart from react-docgen, I also think it's less ergonomic in this case